onnxruntime/core/framework/allocatormgr.h

```diff
@@ -6,11 +6,17 @@
 #include "core/common/common.h"
 #include "core/framework/allocator.h"
 #include "core/session/onnxruntime_c_api.h"
+#include <map>
 
 namespace onnxruntime {
 
 using AllocatorFactory = std::function<std::unique_ptr<IAllocator>(OrtDevice::DeviceId)>;
 
+using AllocatorMap = std::map<int, AllocatorPtr>;
+using MemoryInfoSet = std::set<OrtMemoryInfo>;
```

Comment:
>set [](start = 27, length = 3)

prefer unordered_set instead of set, if no particular sorting stability requirement

Reply:
The code is copied from execution_provider.h, it's duplicate for now. Will this one as the only place to keep the allocators for all EPs in next PR.

---
In reply to: [564297657](https://github.com/microsoft/onnxruntime/pull/6332#discussion_r564297657) [](ancestors = 564297657)

Reply:
will leave it to the next PR as I need to change the OrtMemoryInfo to remove the OrtMemType

---
In reply to: [564717693](https://github.com/microsoft/onnxruntime/pull/6332#discussion_r564717693) [](ancestors = 564717693,564297657)